### PR TITLE
#26 add testing for experts

### PR DESCRIPTION
**Pull request by tania833**
develop

## GitHub Board

**Issue link**

[#26  ]( )

**Story link**

[#26 ]( )

## Code reviewers
- [ ] @github_username

## Summary of issue

- [ ]  Summary of issue
- [ ]  Summary of issue

## Summary of change

- [ ]  Summary of change
- [ ]  Summary of change

### CODE REVIEW

File: src/modules/main/components/__test__/MainExpertsView.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
```
Comment (SergeOlabin) [inline src/modules/main/components/__test__/MainExpertsView.spec.tsx:1]: why need that ?

File: src/modules/main/components/__test__/MainExpertsView.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+describe('MainExpertsView', () => {
+  it('renders images', () => {
+    render(
+      <Provider store={store}>
+        <MainExpertsView />
+      </Provider>,
+    );
+    const images = screen.queryAllByAltText('doctor');
+    expect(images).toHaveLength(11);
```
Comment (SergeOlabin) [inline src/modules/main/components/__test__/MainExpertsView.spec.tsx:16]: why exactly 11 ? :)

Comment (SergeOlabin) [reply src/modules/main/components/__test__/MainExpertsView.spec.tsx:16]: expect for the test must be determined/calculated depending on the test input

File: tsconfig.eslint.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
     "src/**/*.js",
     "src/**/*.jsx",
     "test/**/*.ts",
-    ".eslintrc.js"
+    ".eslintrc.js",
+    "./babel.config.js"
```
Comment (SergeOlabin) [inline tsconfig.eslint.json:10]: why introducing babel?

File: tsconfig.json [hunk 4]
```diff
@@ -27,7 +27,7 @@
     "noImplicitAny": false
   },
   "include": [
-    "src"
+    "src", "babel.config.js"
```
Comment (SergeOlabin) [inline tsconfig.json:30]: why introducing babel?

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 4 above)
Comment (tania833) [reply tsconfig.json]: deleted

File: src/modules/main/components/__test__/MainExpertsView.spec.tsx (hunk 2 above)
Comment (tania833) [reply src/modules/main/components/__test__/MainExpertsView.spec.tsx]: Fixed

**Review by SergeOlabin — COMMENTED** [review]
can we handle tests by means built-in jest only?

File: package.json [hunk 5]
```diff
@@ -49,17 +48,21 @@
     ]
   },
   "devDependencies": {
-    "eslint": "^7.12.1",
-    "prettier": "2.1.2",
+    "@types/jest": "^26.0.15",
     "@typescript-eslint/eslint-plugin": "^4.6.0",
+    "babel-jest": "^26.6.0",
```
Comment (SergeOlabin) [inline package.json:53]: why need babel ? :)

File: src/lib/components/ExpertBlock.tsx [hunk 6]
```diff
@@ -13,6 +13,7 @@ export const ExpertBlock: React.FC<IExpertProps> = (props) => {
   return (
     <>
       <img
+        data-testid='expertBlock'
```
Comment (SergeOlabin) [inline src/lib/components/ExpertBlock.tsx:16]: you don't have to touch the components with testing data.
try decorating the components with the necessary attributes inside the test inself

**Review by SergeOlabin — APPROVED** [review]
cool! let's wait for action pipeline to be green :)

**Pull request merged**